fem/hyperbolic.hpp

```diff
@@ -61,58 +61,102 @@ class FluxFunction
    const int dim;
 
    FluxFunction(const int num_equations, const int dim)
-      : num_equations(num_equations), dim(dim)
-   {
-#ifndef MFEM_THREAD_SAFE
-      flux.SetSize(num_equations, dim);
-#endif
-   }
+      : num_equations(num_equations), dim(dim) { }
+
+   virtual ~FluxFunction() {}
 
    /**
-    * @brief Compute flux F(u, x) for given state u and physical point x
+    * @brief Compute flux. Must be implemented in a derived class.
     *
-    * @param[in] state value of state at the current integration point
-    * @param[in] Tr element information
-    * @param[out] flux F(u, x)
+    * @param[in] state state at the current integration point (num_equations)
+    * @param[in] Tr element transformation
+    * @param[out] flux flux from the given element at the current
+    * integration point (num_equations, dim)
     * @return real_t maximum characteristic speed
     *
     * @note One can put assertion in here to detect non-physical solution
     */
    virtual real_t ComputeFlux(const Vector &state, ElementTransformation &Tr,
                               DenseMatrix &flux) const = 0;
    /**
-    * @brief Compute normal flux. Optionally overloaded in the
-    * derived class to avoid creating full dense matrix for flux.
+    * @brief Compute normal flux. Optionally overloaded in a derived class to
+    * avoid creating a full dense matrix for flux.
     *
-    * @param[in] state state at the current integration point
-    * @param[in] normal normal vector, @see CalcOrtho
-    * @param[in] Tr face information
+    * @param[in] state state at the current integration point (num_equations)
+    * @param[in] normal normal vector, see mfem::CalcOrtho() (dim)
+    * @param[in] Tr face transformation
     * @param[out] fluxDotN normal flux from the given element at the current
-    * integration point
+    * integration point (num_equations)
     * @return real_t maximum (normal) characteristic velocity
     */
    virtual real_t ComputeFluxDotN(const Vector &state, const Vector &normal,
                                   FaceElementTransformations &Tr,
                                   Vector &fluxDotN) const;
 
    /**
-    * @brief Compute flux Jacobian. Optionally overloaded in the derived class
+    * @brief Compute average flux for over the given interval of states.
+    * Optionally overloaded in a derived class.
+    *
+    * @param[in] state1 state of the beggining of the interval (num_equations)
+    * @param[in] state2 state of the end of the interval (num_equations)
+    * @param[in] Tr element transformation
+    * @param[out] flux_ average flux from the given element at the current
+    * integration point (num_equations, dim)
+    * @return real_t maximum characteristic speed
```

Comment:
Is this now the maximum characteristic speed 
- across the interval
- between the two provided states
?

Reply:
With the assumption of monotonicity, these two definitions coincide, so I do not know if to specify. It might get more unclear than clear 😄

Reply:
Maybe add that the flux is assumed monotonic for this method?

Reply:
It is not assumed here, so I specified that it should be maximum over the interval in very general 😉 .